recipes-support/provisioning/files/phytec-board-config.sh

```diff
@@ -0,0 +1,414 @@
+}
+
+set_onboarded() {
+    FILE="${AWSCONFIG_CONFPATH}/${ESECCONFIG}"
+    jsontxt=$(cat ${FILE})
+    jsontxt=$(echo ${jsontxt} | jq --arg para state --arg val "$1" '.onboarding[$para]  = $val')
+    vdate=$(date)
+    jsontxt=$(echo ${jsontxt} | jq --arg para timestamp --arg val "$vdate" '.onboarding[$para] =   $val')
+    echo ${jsontxt} | jq . > ${FILE}
+    set_awsclient start
+}
+
+set_awsclient(){
+    FILE="${AWSCONFIG_CONFPATH}/${ESECCONFIG}"
+    jsontxt=$(cat ${FILE})
+    jsontxt=$(echo ${jsontxt} | jq --arg val "$1" '.awsclient  = $val')
+    echo ${jsontxt} | jq . > ${FILE}
+}
+
+do_opensslconfig() {
+    mkdir -p ${AWSCONFIG_CONFPATH}
+export OPENSSL_CONF=${AWSCONFIG_CONFPATH}/openssl_curl.cnf
+cat > ${AWSCONFIG_CONFPATH}/openssl_curl.cnf <<EOF
+openssl_conf = openssl_init
+
+[openssl_init]
+engines = engine_section
+
+[engine_section]
+pkcs11 = pkcs11_section
+
+[pkcs11_section]
+engine_id = pkcs11
+MODULE_PATH = /usr/lib/libtpm2_pkcs11.so.0
+PIN = my_pin
+init = 0
+EOF
+}
+
+do_clearAWSCONFIG() {
+    rm -r ${AWSCONFIG_PATH}
+}
+
+do_initAWSCONFIG(){
+    do_opensslconfig
+    do_extractcert
+    do_awsconfig
+    return 0
+}
+
+do_resetAWSCONFIG() {
+    do_clearAWSCONFIG
+    do_initAWSCONFIG
+    return 0
+}
+
+do_onboarding() {
+    awsrequest="https://devices.aws.esec-experts.com/ownership/token"
+    if [ $# -eq 1 ]; then
+        awsrequest="--request POST https://devices.aws.esec-experts.com/ownership/customer_account?customer_mail=$1"
+    else
+        do_contract
+        if [ "$ESECCONTRACT" = False ]; then
+            return 0
+        fi
+    fi
+    do_initAWSCONFIG
+    PRIVATEKEYURI="pkcs11:model=SLB9670;manufacturer=Infineon;token=iotdm;object=iotdm-keypair;type=private"
+    response=$(curl --cacert ${AWSCONFIG_CERTPATH}/rootcert.pem --engine pkcs11 --key-type ENG --key "${PRIVATEKEYURI};pin-value=${TPM_PIN}" --cert ${AWSCONFIG_CERTPATH}/devcert.pem ${awsrequest})
+    if [ $? -ne 0 ]; then
+       return 1
+    fi
+    if [ "$INTERACTIVE" = True ]; then
+        #error check
+        check=$(echo ${response} | jq .message)
+        if [ "$check" = "\"Forbidden\"" ]; then
+            return 2
+        fi
+        check=$(echo ${response} | jq .errorType)
+        if [ ! -z "$check" ] && [ ! "$check" = null ]; then
+            check=$(echo ${response} | jq .errorMessage)
+            whiptail --msgbox "\
+Error Message from IoT Device Manager
+$check" $WT_HEIGHT $WT_WIDTH
+            return 0
+        fi
+        if [ $# -eq 0 ]; then
+            # get token
+            set_onboarded tokendevice
+            certserial=$(echo ${response} | jq .thing_name)
+            token=$(echo ${response} | jq .token)
+            valid=$(echo ${response} | jq .valid_until)
+            whiptail --msgbox "\
+Welcome to the ESEC IoT Device Manager Platform
+The next steps:
+ 1) Login to your account on the ESEC IoT Device
+    Manager Platform https://iot.aws.esec-experts.com.
+    If you have no account,
```

Comment:
"If you do not have an account yet, then use option 1) New Account and Onboarding to register."